app/modules/ads1115.c

```diff
@@ -0,0 +1,377 @@
+	platform_i2c_send_start(ads1115_i2c_id);
+	platform_i2c_send_address(ads1115_i2c_id, ads1115_i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(ads1115_i2c_id, reg);
+	platform_i2c_send_byte(ads1115_i2c_id, (uint8_t)(config >> 8));
+	platform_i2c_send_byte(ads1115_i2c_id, (uint8_t)(config & 0xFF));
+	platform_i2c_send_stop(ads1115_i2c_id);
+}
+
+static uint16_t read_reg(uint8_t reg) {
+	platform_i2c_send_start(ads1115_i2c_id);
+	platform_i2c_send_address(ads1115_i2c_id, ads1115_i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(ads1115_i2c_id, reg);
+	platform_i2c_send_stop(ads1115_i2c_id);
+	platform_i2c_send_start(ads1115_i2c_id);
+	platform_i2c_send_address(ads1115_i2c_id, ads1115_i2c_addr, PLATFORM_I2C_DIRECTION_RECEIVER);
+	uint16_t buf = (platform_i2c_recv_byte(ads1115_i2c_id, 1) << 8);
+	buf += platform_i2c_recv_byte(ads1115_i2c_id, 0);
+	platform_i2c_send_stop(ads1115_i2c_id);
+	return buf;
+}
+
+// convert ADC value to voltage corresponding to PGA settings
+static double get_volt(uint16_t value) {
+	
+	double volt = 0;
+	
+	switch (ads1115_gain) {
+		case (ADS1115_PGA_6_144V):
+			volt = (int16_t)value * 0.1875;
+			break;
+		case (ADS1115_PGA_4_096V):
+			volt = (int16_t)value * 0.125;
+			break;
+		case (ADS1115_PGA_2_048V):
+			volt = (int16_t)value * 0.0625;
+			break;
+		case (ADS1115_PGA_1_024V):
+			volt = (int16_t)value * 0.03125;
+			break;
+		case (ADS1115_PGA_0_512V):
+			volt = (int16_t)value * 0.015625;
+			break;
+		case (ADS1115_PGA_0_256V):
+			volt = (int16_t)value * 0.0078125;
+			break;
+	}
+	
+	return volt;
+}
+
+// convert threshold in volt to ADC value corresponding to PGA settings
+static uint8_t get_value(int16_t *volt) {
+	
+	switch (ads1115_gain) {
+		case (ADS1115_PGA_6_144V):
+			if ((*volt >= 6144) || (*volt < -6144) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.1875;
+			break;
+		case (ADS1115_PGA_4_096V):
+			if ((*volt >= 4096) || (*volt < -4096) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.125;
+			break;
+		case (ADS1115_PGA_2_048V):
+			if ((*volt >= 2048) || (*volt < -2048) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.0625;
+			break;
+		case (ADS1115_PGA_1_024V):
+			if ((*volt >= 1024) || (*volt < -1024) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.03125;
+			break;
+		case (ADS1115_PGA_0_512V):
+			if ((*volt >= 512) || (*volt < -512) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.015625;
+			break;
+		case (ADS1115_PGA_0_256V):
+			if ((*volt >= 256) || (*volt < -256) || ((*volt < 0) && (ads1115_mode >> 2))) return 1;
+			*volt = *volt / 0.0078125;
+			break;
+	}
+	
+	return 0;
+}
+
+// wait for ADC conversion corresponding to data rate settings
+static uint16_t conversion_delay(void) {
+	
+	switch (ads1115_samples) {
+		case (ADS1115_DR_8SPS):
+			os_delay_us(128000);
```

Comment:
Maybe this should use a timer instead of a delay. A current issue (#1892) is suggesting that we remove all instances of os_delay_us from the code base.